manual/src/server/sal.md

```diff
@@ -0,0 +1,789 @@
+# SAL
+
+## Tfgrid
+
+### Load
+- loads key
+- chooses network
+- starts session and creates state for client
+
+```
+func (c *Client) Load(args Load) error
+```
+
+### DeployVM
+- Deploy a single VM 
+- Choose node id, farm id, etc 
+
+```
+func (c *Client) DeployVM(args tfgridBase.VMDeployment) (tfgridBase.VMDeployment, error)
+```
+
+### GetVMDeployment
+- Retrieve information about the deployed vm
+
+```
+func (c *Client) GetVMDeployment(name string) (tfgridBase.VMDeployment, error)
+```
+
+### CancelVMDeployment
+- Cancel the vm deployment
+
+```
+func (c *Client) CancelVMDeployment(name string) error
+```
+
+### DeployNetwork
+- Deploy VMs that will be in the same network
+- Provide name and network specs
+- Provide the VM specs (on what node, farmid, flist, etc.)
+
+```
+func (c *Client) DeployNetwork(args tfgridBase.NetworkDeployment) (tfgridBase.NetworkDeployment, error)
+```
+
+### GetNetworkDeployment
+- Retrieve information about the vms you deployed by providing the name when deploying
+```
+func (c *Client) GetNetworkDeployment(name string) (tfgridBase.NetworkDeployment, error)
+```
+
+### CancelNetworkDeployment
+- Delete all deployed VMs that are part of the model that goes by the modelName
+
+```
+func (c *Client) CancelNetworkDeployment(name string) error
+```
+
+### ExtendNetworkDeployment
+- Add a VM to the network
+```
+func (c *Client) ExtendNetworkDeployment(args tfgridBase.ExtendNetworkDeployment) (tfgridBase.NetworkDeployment, error)
+```
+
+### ShrinkNetworkDeployment
```

Comment:
can't we use DetachNeworkToVm as a name?

Reply:
Same here. The function cancels the VM from that network. So it does indeed detach it but also cancels it.

Reply:
Hmm, you're right indeed, but I still think it's the network spreading over zos nodes, not just the VMs? @MarioBassem can you still confirm please?

Reply:
like above, the calls that achieve this are:
-  https://github.com/threefoldtech/web3_proxy/blob/13a10a86572569dbe8b780c91a3f28a86d375489/server/pkg/tfgrid/client.go#L118
- https://github.com/threefoldtech/web3_proxy/blob/13a10a86572569dbe8b780c91a3f28a86d375489/server/pkg/tfgrid/client.go#L163
- https://github.com/threefoldtech/web3_proxy/blob/13a10a86572569dbe8b780c91a3f28a86d375489/server/pkg/tfgrid/solutions.go#L173

they delete the proper workloads or deployments, and if the network workload on a node that had updates is not used, it gets deleted too.

Reply:
@xmonader what do you mean by the network that is spreading over zos nodes?